### ISSUE TITLE
Add option to generally use display serial number instead of CGDirectDisplayID for preferences

### ISSUE THREAD

**Issue by waydabber** [MonitorControl/MonitorControl#961]
### Discussed in https://github.com/MonitorControl/MonitorControl/discussions/960

With Apple Silicon `CGDirectDisplayID` is not as unique as it was with Intel but a simple number that is related to the connection's location in the IO registry. The problem is that MonitorControl stores preferences attached to this ID which can cause unexpected things especially when two displays with very different characteristics are being connected to the same port at successive occasions. There should be an option to use display serial number for associating preferences instead of `CGDirectDisplayID`.

**Comment by waydabber** [comment]
Also see this one - https://github.com/MonitorControl/MonitorControl/issues/792

**Comment by waydabber** [comment]
All right... I was wrong, I checked the code again and realized that I actually already changed how preferences ID works in MC, since 4.0.0 the preferences ID also contains data from display name, vendor and model number as a composite ID and this is what is tied to a specific display's stored settings (the ID should be something like `(LGUltraHD778923305@2)` where only the `@2` is the port/ID part and all the stuff before that contains other things that it makes the display unique). :] This means it should not get confused between the two displays, that was a problem in 3.0.0 only.

I'll close this one.
